tests/conftest.py

```diff
@@ -922,33 +922,51 @@ def create_mock_mqtt(*args, **kwargs) -> MqttMockHAClient:
 
 @pytest.fixture
 def hass_config() -> ConfigType | None:
-    """Fixture to override configuration.yaml using mock_yaml_config.
+    """Fixture to parameterize the content of configuration.yaml using mock_yaml_config.
 
-    Fixture to parameterize the content of configuration.yaml with the
-    yaml dump of a config dict.
+    To set a configuration, tests can be marked with:
+    @pytest.mark.parametrize("hass_config", [{integration: {}}])
+    and the `mock_hass_config: None` fixture to the test parameters.
     """
     return None
 
 
 @pytest.fixture
-def hass_config_yaml(hass_config: ConfigType | None) -> str | None:
-    """Fixture to override configuration.yaml using mock_yaml_config.
+def mock_hass_config(
+    hass: HomeAssistant, hass_config: ConfigType | None
+) -> Generator[None, None, None]:
+    """Fixture to mock the content of configuration.yaml.
+
+    Patches configuration files with the yaml dump of `hass_config` as content.
+    """
+    with patch_yaml_files({YAML_CONFIG_FILE: yaml.dump(hass_config)}):
+        yield
```

Comment:
Can this path not patch `homeassistant.config.async_hass_config_yaml` directly, to avoid yaml dump+parse?

```suggestion
    with patch(
        "homeassistant.config.async_hass_config_yaml", return_value=hass_config
    )
        yield
```

Reply:
I just noticed that async_hass_config_yaml will skip validation.
Maybe instead patch `homeassistant.config.load_yaml_config_file`